### DISCUSSION TITLE
Which characters should be only Unicoded

### DISCUSSION THREAD

**Discussion by mariannevanharten** [FamilySearch/GEDCOM#523] [Q&A] [answered]
In which range are the characters that should be Unicoded?
So, in which range I may not to use the regular character?

Best regards,
Marianne van Harten

**Comment by tychonievich** [comment]
GEDCOM 7 files are encoded in UTF-8 in their entirety. UTF-8 encodes code points 0 through 127 as a single byte with that code point as its numerical value, while other code points are encoded as multi-byte sequences.

UTF-8 is one of the most widely supported character encodings in the world: it is likely that there is library support for it in whatever language you are using (or a setting for it in your editor if you are typing GEDCOM by hand). I've written multiple GEDCOM files and GEDCOM-handling applications without doing anything more to handle UTF-8 than selecting it as the encoding of the file.

**Comment by mariannevanharten** [comment]
Am I right to say that U+00xx can be replaced by its regular character?

**Reply by tychonievich** [reply]
I'm not sure what you mean by "regular character." If you you mean something like C's `char` type (a single 8-bit byte) then that only works for U+0000 through U+007F, not for U+0080 or larger code points.

**Reply by mariannevanharten** [reply]
I think that you already helped me.
I want to know in which range I can use characters such as ë, á, etc. and when I SHOULD use Unicode.

**Comment by fisharebest** [comment, marked as answer]
Hi Marianne.  I just looked at your site, and I think you are mis-understanding Unicode.

You have entered things such as `Alleb%U+00E9` into your data.  i.e. the literal characters `U`, `+`, `0`, `0`, `E`, `9`.

`U+0039` is just an unambigious, human-friendly way to refer to the unicode character `é`.

You should **always** be using the actual characters in your data, such as `é`.

No encoding is necessary.  These will "just work".

**Comment by mariannevanharten** [comment]
Hi Greg,

 

You are right.

Last year I changed some characters to the Unicode format in the beginning of Gedcom 7. Now I’ve changed most of the Unicode characters back to the literal one.

When checking my gedcom using Chronoplex’s Gedcom Validator, I see many good old 5.5.1 tags (which are equivalent to the 7.0 tag). Except these “errors”, a few other errors remain related to Gedcom 7… 😊

 

 

Vriendelijke groet,
best regards, 

  <https://vanharten.net/wapen.jpg> 

Marianne van Harten

Van: Greg Roach ***@***.***> 
Verzonden: maandag 29 juli 2024 17:58
Aan: FamilySearch/GEDCOM ***@***.***>
CC: Marianne van Harten ***@***.***>; Author ***@***.***>
Onderwerp: Re: [FamilySearch/GEDCOM] Which characters should be only Unicoded (Discussion #523)

 

Hi Marianne. I just looked at your site, and I think you are mis-understanding Unicode.

You have entered things such as Alleb%U+00E9 into your data. i.e. the literal characters U, +, 0, 0, E, 9.

U+0039 is just an unambigious, human-friendly way to refer to the unicode character é.

You should always be using the actual characters in your data, such as é.

No encoding is necessary. These will "just work".
